packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1599,9 +1645,14 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
         child: barrier,
       );
     }
-    barrier = IgnorePointer(
-      ignoring: animation!.status == AnimationStatus.reverse || // changedInternalState is called when animation.status updates
-                animation!.status == AnimationStatus.dismissed, // dismissed is possible when doing a manual pop gesture
+    barrier = AnimatedBuilder(
+      animation: _ignorePointerNotifier,
+      builder: (BuildContext context, Widget? child) {
+        return IgnorePointer(
+          ignoring: _ignorePointer,
+          child: child,
+        );
+      },
```

Comment:
This makes the interactivity of modal barriers during transitions follow the same rules as interactivity for full screen routes